File: LAB2/C/Goal 1/main.c
#include <stdio.h>
#include <stdbool.h>

int main () {
        double Sum = 0;
        int i, N = 0, PositiveCount = 0;
        bool InValid = true;

        printf("\nThis program calculates arithmetic mean of all positive elements.\n\n");
        do {
            printf("Enter number of elements (<21):\n");
            scanf("%d", &N);
            if ( N > 0) {
                InValid = false;
            } else {
                printf("Invalid value. Try again\n");
            }
        } while (InValid);
        double A[N];
        for (i = 0; i < N; i++) {
            printf("Enter element A[%d] (absolute value not bigger than 100 000 000):\n", i + 1);
            scanf("%lf", &A[i]);
        }
        for ( i = 0; i < N; i++) {
            if (A[i] > -1) {
                Sum = Sum + A[i];
                PositiveCount++;
            }
        }
        Sum /= PositiveCount;
        printf("%.3f - ANSWER", Sum);
}

File: LAB2/Java/Goal 1/src/Main.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        final int MAX_LENGTH = 21;
        final int MAX_SIZE = 100000001;
        final int MIN_SIZE = -100000001;

        double Sum = 0;
        int i = 0, N = 0, PositiveCount = 0;
        boolean InValid = true;

        System.out.println("This program calculates arithmetic mean of all positive elements.\n");

        do {
            try {
                System.out.println("Enter number of elements (<" + MAX_LENGTH + "): ");
                N = Integer.parseInt(in.nextLine());
                if ((N > 0) && (N < MAX_LENGTH)) {
                    InValid = false;
                } else {
                    System.out.println("Error! Value is not valid.");
                }
            } catch (Exception e) {
                System.out.println("INPUT ERROR!");
            }
        } while (InValid);

        double[] A = new double[N];

        while (i < N) {
            try {
                System.out.println("Enter element A[" + (i + 1) + "] (absolute value less than " + MAX_SIZE +"):");
                A[i] = Double.parseDouble(in.nextLine());
                if ((A[i] < MAX_SIZE) && (A[i] > MIN_SIZE)) {
                    i++;
                } else {
                    System.out.println("Error! Value is not valid.");
                }
            } catch (Exception e) {
                System.out.println("INPUT ERROR!");
            }
        }
        for ( i = 0; i < N; i++) {
            if (A[i] > -1) {
                Sum = Sum + A[i];
                PositiveCount++;
            }
        }
        Sum /= PositiveCount;
        System.out.printf("%.3f - ANSWER", Sum);
    }
}
